### DISCUSSION TITLE
Split out DLNA code into separate module/repo for reuse in other projects?

### DISCUSSION THREAD

**Discussion by dweymouth** [alexballas/go2tv#59] [Ideas]
Hi, I was wondering if you would potentially be interested in splitting out the DLNA code (device discovery, soapcalls, file server, etc) into another repo so it can be a standalone library for other DLNA casting applications? I am working on a Subsonic music player client and would like to eventually add uPnP/DLNA casting to my app (also written in Go+Fyne). There is also https://github.com/ianr0bkny/go-sonos but it's ancient and unmaintained and I wasn't able to get it to work after some basic testing, so I think the Go ecosystem could benefit from a new uPnP/DLNA client library. I don't know the protocol very well but I'd also hope to add other features I'd need for my app, like seeking within a file.

**Comment by alexballas** [comment]
Hello, Go2TV already exposes a public API which you can find here.

https://pkg.go.dev/github.com/alexballas/go2tv@v1.13.0/devices
https://pkg.go.dev/github.com/alexballas/go2tv@v1.13.0/soapcalls
https://pkg.go.dev/github.com/alexballas/go2tv@v1.13.0/httphandlers
https://pkg.go.dev/github.com/alexballas/go2tv@v1.13.0/utils

For ssdp discovery I use [github.com/koron/go-ssdp](url). Never took the time to implement it.

In theory you could rewrite Go2TV just by using these packages. Also I try to keep the API relatively stable. (there will be some breaking changes with the upcoming v1.14.0 version however)

I don't believe right now that splitting this into a separate standalone library would bring any great benefit. I don't plan to implement the full UPnP AV architecture unless it's something related to Go2TV. Also the demand for a UPNP client is not big right now.
